src/lib/app-helper.js

```diff
@@ -265,43 +254,44 @@ async function downloadOWJar (url, outFile) {
     })
   })
 }
+
+/** @private */
+async function waitForOpenWhiskReadiness (host, endTime, period, timeout, waitFunc) {
+  if (Date.now() > endTime) {
+    throw new Error(`local openwhisk stack startup timed out: ${timeout}ms`)
+  }
```

Comment:
With respect to this function, this just retries until the OW server is ready so we can proceed with the rest of the flow (via a fetch call to `/api/v1`). That is the only spot where I can see if the path does not exist (in the future?), the response would fail (even though we only really need to check whether it is accessible). 

In that case, this should be a generic HTTP call to the root path for the OW server. All OW servers do respond with a json payload with metadata:
```
{
    "api_paths": [
        "/api/v1"
    ],
    "description": "OpenWhisk",
    "limits": {
        "actions_per_minute": 60,
        "concurrent_actions": 30,
        "max_action_duration": 300000,
        "max_action_logs": 10485760,
        "max_action_memory": 536870912,
        "min_action_duration": 100,
        "min_action_logs": 0,
        "min_action_memory": 134217728,
        "sequence_length": 50,
        "triggers_per_minute": 60
    },
    "runtimes": {
        "nodejs": [
            {
                "attached": false,
                "default": true,
                "deprecated": false,
                "image": "adobeapiplatform/adobe-action-nodejs-v12:3.0.22",
                "kind": "nodejs:12",
                "requireMain": false
            },
            {
                "attached": false,
                "default": false,
                "deprecated": false,
                "image": "adobeapiplatform/adobe-action-nodejs-v10:3.0.22",
                "kind": "nodejs:10",
                "requireMain": false
            }
        ]
    },
    "support": {
        "github": "https://github.com/apache/openwhisk/issues",
        "slack": "http://slack.openwhisk.org"
    }
}
```

I'll add an issue for this.

Reply:
Filed https://github.com/adobe/aio-cli-plugin-app/issues/363